### PR TITLE
Vandalism

### LINKED ISSUE #212: Prevent (obvious) vandalism
Mark 'public' on the feed row for whether the feed should appear on the browse page.

1. Add column to feed table that is "public" that can only be changed by an admin (set up #120)
2. Auto-manage banned list with some basic regex parsing for bad words in the feed title or description (e.g. porn), when a new feed is added to the system

### PR DESCRIPTION
**Pull request by alexgerstein**
Closes #212.

### CODE REVIEW

File: json/bad_words.json [hunk 1]
```diff
@@ -0,0 +1 @@
... "penisfucker", "phonesex", "phuck", "phuk", "phuked", "phuking", "phukked", "phukking", "phuks", "phuq", "pigfucker", "pimpis", "piss", "pissed", "pisser", "pissers", "pisses", "pissflaps", "pissin", "pissing", "pissoff", "poop", "porn", "porno", "pornography", "pornos", "prick", "pron", "pube", "pusse", "pussi", "pussies", "pussy", "rectum", "retard", "rimjaw", "rimming", "s.o.b.", "sadist", "schlong", "screwing", "scroat", "scrote", "scrotum", "semen", "sex", "sh!\\+", "sh!t", "sh1t", "shag", "shagger", "shaggin", "shagging", "shemale", "shi\\+", "shit", "shitdick", "shite", "shited", "shitey", "shitfuck", "shitfull", "shithead", "shiting", "shits", "shitted", "shitter", "shitting", "shitty", "skank", "slut", "sluts", "smegma", "smut", "snatch", "son-of-a-bitch", "spac", "spunk", "s_h_i_t", "t1tt1e5", "t1tties", "teets", "teez", "testical", "testicle", "tit", "titfuck", "tits", "titt", "tittie5", "tittiefucker", "titties", "tittyfuck", "tittywank", "titwank", "tosser", "turd", "tw4t", "twat", "twathead", "twatty", "twunt", "twunter", "v14gra", "v1gra", "vagina", "viagra", "vulva", "w00se", "wang", "wank", "wanker", "wanky", "whoar", "whore", "willies", "willy", "xrated", "xxx"]}
```
Comment (swglad) [inline json/bad_words.json]: Add comment with link to source (where'd you find this list)

File: sourcemash/api/feeds.py [hunk 2]
```diff
@@ -15,6 +17,9 @@
 from worker.scraper import scrape_feed_articles
 
 REDIS_CONNECTION = create_worker()
+BAD_WORDS_FILE = "./json/bad_words.json"
+print os.path.realpath(BAD_WORDS_FILE)
```
Comment (swglad) [inline sourcemash/api/feeds.py]: Delete.

File: sourcemash/api/feeds.py [hunk 3]
```diff
@@ -74,6 +79,14 @@ def post(self):
         if rss_feed['bozo'] == 1:
             return {"errors": {"url": ["URL is not a valid feed"]}}, 422
 
+        with open(BAD_WORDS_FILE) as data_file:
+            data = json.load(data_file)
+            badwords = data['badwords']
+            words = rss_feed['feed']['title'] + rss_feed['feed']['description']
```
Comment (swglad) [inline sourcemash/api/feeds.py]: Could this lead to a weird middle world?

E.g. "I want to dance." + "Hello, World!" --> "I want to dance.Hello, World!"

Comment (swglad) [inline sourcemash/api/feeds.py]: Fix: ```words = rss_feed['feed']['title'] + " " + rss_feed['feed']['description']```

File: sourcemash/api/feeds.py [hunk 4]
```diff
@@ -109,7 +123,7 @@ def post(self):
 class FeedListAllAPI(Resource):
 
     def get(self):
-        return {'feeds': [marshal(feed, feed_status_fields) for feed in Feed.query.all()]}
+        return {'feeds': [marshal(feed, feed_status_fields) for feed in Feed.query.filter(Feed.public==True).all()]}
```
Comment (swglad) [inline sourcemash/api/feeds.py]: This means that if newly added feeds default to public=False, then they'll never show up on the Browse page for that user. This is a big issue.

File: json/bad_words.json (hunk 1 above)
Comment (alexgerstein) [inline json/bad_words.json]: Done.

File: sourcemash/api/feeds.py [hunk 5]
```diff
@@ -109,7 +129,16 @@ def post(self):
...
 
     def get(self):
-        return {'feeds': [marshal(feed, feed_status_fields) for feed in Feed.query.all()]}
+        feeds = Feed.query.filter(Feed.public).all()
+
+        if current_user.is_authenticated():
+            logger.info("ANONYMOUS USER")
+            logger.info(current_user.subscribed)
+            feeds += current_user.subscribed.filter(Feed.public==False).all()
+        logger.info("NOT ANONYMOUS USER")
```
Comment (swglad) [inline sourcemash/api/feeds.py]: Are these loggings relevant?

File: tests/test_api/test_feeds.py [hunk 6]
```diff
@@ -91,6 +92,27 @@ def test_get_feeds(self, test_client, feed):
...
 
+    def test_get_feeds_public_only(self, test_client, user, private_feed):
+        r = test_client.get('/api/feeds/all')
+        check_valid_header_type(r.headers)
+        assert r.status_code == 200
+
+        data = json.loads(r.data)
+        assert len(data['feeds']) == 0
+
+    def test_get_feeds_if_user_subscribed(self, db, test_client,
```
Comment (swglad) [inline tests/test_api/test_feeds.py]: db?

File: manage.py [hunk 7]
```diff
@@ -60,6 +62,13 @@ def scrape(noqueue=False):
             job = q.enqueue_call(func=scrape_and_categorize_articles, timeout=1800)
             time.sleep(THIRTY_MINUTES)
 
+=======
```
Comment (swglad) [inline manage.py]: watch out.

File: manage.py [hunk 8]
```diff
@@ -60,6 +62,13 @@ def scrape(noqueue=False):
...
+    q = Queue('default', connection=conn)
+    while True:
+        q.enqueue_call(func=scrape_and_categorize_articles, timeout=1800)
+        time.sleep(THIRTY_MINUTES)
+>>>>>>> PR fixes.
```
Comment (swglad) [inline manage.py]: Here too.

File: manage.py [hunk 9]
```diff
@@ -52,6 +53,7 @@ def test(all=False):
 def scrape(noqueue=False):
     """Start an infinte loop to scrape & categorize articles."""
 
+<<<<<<< HEAD
```
Comment (swglad) [inline manage.py]: Here too.

File: sourcemash/api/feeds.py [hunk 10]
```diff
@@ -84,7 +101,7 @@ def post(self):
             feed = Feed(title=rss_feed['feed']['title'],
                         url=rss_feed['url'],
                         description=rss_feed['feed']['description'],
-                        topic=MASH_TOPIC,
+                        public=Falsetopic=MASH_TOPIC,
```
Comment (swglad) [inline sourcemash/api/feeds.py]: Syntax error

File: sourcemash/api/feeds.py [hunk 11]
```diff
@@ -84,7 +101,12 @@ def post(self):
             feed = Feed(title=rss_feed['feed']['title'],
                         url=rss_feed['url'],
                         description=rss_feed['feed']['description'],
+<<<<<<< HEAD
```
Comment (swglad) [inline sourcemash/api/feeds.py]: REbase issues.

**Pull request merged**